sklearn/metrics/_plot/tests/test_plot_confusion_matrix.py

```diff
@@ -0,0 +1,193 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose
+from numpy.testing import assert_array_equal
+
+from sklearn.datasets import make_classification
+from sklearn.svm import SVC, SVR
+from sklearn.metrics import confusion_matrix
+from sklearn.metrics import plot_confusion_matrix
+from sklearn.metrics import ConfusionMatrixDisplay
+
+
+@pytest.fixture(scope="module")
+def n_classes():
+    return 5
+
+
+@pytest.fixture(scope="module")
+def data(n_classes):
+    X, y = make_classification(n_samples=100, n_informative=5,
+                               n_classes=n_classes, random_state=0)
+    return X, y
+
+
+@pytest.fixture(scope="module")
+def fitted_clf(data):
+    return SVC(kernel='linear', C=0.01).fit(*data)
+
+
+@pytest.fixture(scope="module")
+def y_pred(data, fitted_clf):
+    X, _ = data
+    return fitted_clf.predict(X)
+
+
+def test_error_on_regressor(pyplot, data):
+    X, y = data
+    est = SVR().fit(X, y)
+
+    msg = "plot_confusion_matrix only supports classifiers"
+    with pytest.raises(ValueError, match=msg):
+        plot_confusion_matrix(est, X, y)
+
+
+def test_error_on_invalid_option(pyplot, fitted_clf, data):
+    X, y = data
+    msg = (r"normalize must be one of \{'true', 'pred', 'all', "
+           r"None\}")
+
+    with pytest.raises(ValueError, match=msg):
+        plot_confusion_matrix(fitted_clf, X, y, normalize='invalid')
+
+
+@pytest.mark.parametrize("normalize", ['true', 'pred', 'all', None])
+@pytest.mark.parametrize("with_sample_weight", [True, False])
+@pytest.mark.parametrize("with_labels", [True, False])
+@pytest.mark.parametrize("cmap", ['viridis', 'plasma'])
+@pytest.mark.parametrize("with_custom_axes", [True, False])
+@pytest.mark.parametrize("with_display_labels", [True, False])
+@pytest.mark.parametrize("include_values", [True, False])
```

Comment:
Do we really need each of these combinations to be tested independently?

It seems to me that most of the checks in this test could be independent tests functions. Parametrization is nice but seems way overkill here.

This will test 256 instances, and it take about 10s on my machine which is not negligible considering small increment in testing time really add up over time.